src/structures/Channel.js

```diff
@@ -13,7 +13,7 @@ class Channel extends Base {
   constructor(client, data) {
     super(client);
 
-    const type = Object.keys(ChannelTypes)[data.type];
+    const type = Object.keys(ChannelTypes).find(typeName => ChannelTypes[typeName] === data.type);
```

Comment:
At this point, should we consider making ChannelTypes a proper enum? This change makes the line 3 times slower, which has a noticeable impact in bots with hundreds of thousands of channels.